packages/server/lib/plugins/child/cross_origin.js

```diff
@@ -1,22 +1,19 @@
-import { getFullWebpackOptions } from '@cypress/webpack-batteries-included-preprocessor'
-import md5 from 'md5'
-import { fs } from 'memfs'
-import * as path from 'path'
-import webpack from 'webpack'
-
+const md5 = require('md5')
+const { fs } = require('memfs')
+const path = require('path')
+const webpack = require('webpack')
 const VirtualModulesPlugin = require('webpack-virtual-modules')
 
-interface Options {
-  file: string
-  fn: string
-}
+const resolve = require('../../util/resolve')
 
-// @ts-expect-error - webpack expects `fs.join` to exist for some reason
 fs.join = path.join
 
-export const processCallback = ({ file, fn }: Options) => {
+const processCallback = ({ file, fn, projectRoot }) => {
+  const { getFullWebpackOptions } = require('@cypress/webpack-batteries-included-preprocessor')
+
   const source = fn.replace(/Cypress\.require/g, 'require')
-  const webpackOptions = getFullWebpackOptions(file, require.resolve('typescript'))
+  const typescriptPath = resolve.typescript(projectRoot)
+  const webpackOptions = getFullWebpackOptions(file, typescriptPath)
```

Comment:
This is the root of the fix. Instead of resolving typescript with our own, we get the path to it the same way we do for regular file preprocessing by finding the user's installed typescript.